File: tests/algorithms/x/Java/greedy_methods/gas_station.java
public class Main {
    static class GasStation {
        int gas_quantity;
        int cost;
        GasStation(int gas_quantity, int cost) {
            this.gas_quantity = gas_quantity;
            this.cost = cost;
        }
        GasStation() {}
        @Override public String toString() {
            return String.format("{'gas_quantity': %s, 'cost': %s}", String.valueOf(gas_quantity), String.valueOf(cost));
        }
    }

    static GasStation[] example1;
    static GasStation[] example2;

    static GasStation[] get_gas_stations(int[] gas_quantities, int[] costs) {
        GasStation[] stations = ((GasStation[])(new GasStation[]{}));
        int i = 0;
        while (i < gas_quantities.length) {
            stations = ((GasStation[])(java.util.stream.Stream.concat(java.util.Arrays.stream(stations), java.util.stream.Stream.of(new GasStation(gas_quantities[i], costs[i]))).toArray(GasStation[]::new)));
            i = i + 1;
        }
        return stations;
    }

    static int can_complete_journey(GasStation[] gas_stations) {
        int total_gas = 0;
        int total_cost = 0;
        int i_1 = 0;
        while (i_1 < gas_stations.length) {
            total_gas = total_gas + ((Number)(((Object)(((java.util.Map)gas_stations[i_1])).get("gas_quantity")))).intValue();
            total_cost = total_cost + ((Number)(((Object)(((java.util.Map)gas_stations[i_1])).get("cost")))).intValue();
            i_1 = i_1 + 1;
        }
        if (total_gas < total_cost) {
            return -1;
        }
        int start = 0;
        int net = 0;
        i_1 = 0;
        while (i_1 < gas_stations.length) {
            GasStation station = gas_stations[i_1];
            net = net + (int)(((int)(((java.util.Map)station)).getOrDefault("gas_quantity", 0))) - (int)(((int)(((java.util.Map)station)).getOrDefault("cost", 0)));
            if (net < 0) {
                start = i_1 + 1;
                net = 0;
            }
            i_1 = i_1 + 1;
        }
        return start;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            example1 = ((GasStation[])(get_gas_stations(((int[])(new int[]{1, 2, 3, 4, 5})), ((int[])(new int[]{3, 4, 5, 1, 2})))));
            System.out.println(_p(can_complete_journey(((GasStation[])(example1)))));
            example2 = ((GasStation[])(get_gas_stations(((int[])(new int[]{2, 3, 4})), ((int[])(new int[]{3, 4, 3})))));
            System.out.println(_p(can_complete_journey(((GasStation[])(example2)))));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/PHP/greedy_methods/gas_station.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
$__start_mem = memory_get_usage();
$__start = _now();
  function get_gas_stations($gas_quantities, $costs) {
  global $example1, $example2;
  $stations = [];
  $i = 0;
  while ($i < count($gas_quantities)) {
  $stations = _append($stations, ['gas_quantity' => $gas_quantities[$i], 'cost' => $costs[$i]]);
  $i = $i + 1;
};
  return $stations;
};
  function can_complete_journey($gas_stations) {
  global $example1, $example2;
  $total_gas = 0;
  $total_cost = 0;
  $i = 0;
  while ($i < count($gas_stations)) {
  $total_gas = $total_gas + $gas_stations[$i]['gas_quantity'];
  $total_cost = $total_cost + $gas_stations[$i]['cost'];
  $i = $i + 1;
};
  if ($total_gas < $total_cost) {
  return -1;
}
  $start = 0;
  $net = 0;
  $i = 0;
  while ($i < count($gas_stations)) {
  $station = $gas_stations[$i];
  $net = $net + $station['gas_quantity'] - $station['cost'];
  if ($net < 0) {
  $start = $i + 1;
  $net = 0;
}
  $i = $i + 1;
};
  return $start;
};
  $example1 = get_gas_stations([1, 2, 3, 4, 5], [3, 4, 5, 1, 2]);
  echo rtrim(_str(can_complete_journey($example1))), PHP_EOL;
  $example2 = get_gas_stations([2, 3, 4], [3, 4, 3]);
  echo rtrim(_str(can_complete_journey($example2))), PHP_EOL;
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;
